docs/experiments/taint-propagators.md

```diff
@@ -0,0 +1,53 @@
+---
+slug: taint-propagators
+append_help_link: true
+description: "Taint propagators allow you to customize how taint is propagated."
+---
+
+# Taint propagators
+
+Consider the following Python code:
+
+```python
+def test(s):
+    x = user_input
+    s = set([])
+    s.add(x)
+    #ruleid: test
+    sink(s.pop())
+```
+
+Here, an unsafe `user_input` is stored into a set data-structure, and then we take
+a random element from that set and pass it into a `sink` function. Of course, this
+random element could be `user_input` itself, leading to an injection vulnerability!
+
+If we try to find this issue with this rule:
+
+```yaml
+mode: taint
+pattern-sources:
+- pattern: user_input
+pattern-sinks:
+- pattern: sink(...)
+```
+
+It will not find it. This is because, by itself, Semgrep only propagates taint
+through assignments.
+
+Using _taint propagators_ you can tell Semgrep how to propagate taint in these
+scenarios. Taint propagators are specified under the `pattern-propagators` key:
+
+```yaml
+pattern-propagators:
```

Comment:
@IagoAbal I was a bit confused by naming - how do we want to call this feature? Is it:

- Pattern propagator (as mentioned in the code snippet here)
- Taint propagator (as in the header)
- Propagator (as we call it in some paragraphs)

We can find one name and stick to it.

Reply:
The feature is "taint propagators" but the operator is `pattern-propagators` because all our operators start with `pattern-`. That we cannot change. Now, "propagator" is just a short form of "taint propagator". We could rewrite those if you insist.

Reply:
@IagoAbal thanks for the explanation, looks like you fixed it!